### ISSUE TITLE
Too generic content type for JSON messages via AMQP

### ISSUE THREAD

**Issue by isbm** [openSUSE/open-build-service#10147]
# Issue Description

Content type for JSON messages via AMQP mainly "application/octet-stream" and sometimes "application/json".

# Expected Result

JSON messages should have "application/json" content type.

**Comment by hennevogel** [comment]
See the explanation in #10146

**Comment by isbm** [comment]
@hennevogel  Huh? Before you go and randomly close unresolved issues, you should a bit try to understand the problem!

Content type is there to indicate the media type of the resource. And if OBS puts different resources with the same content type, one should already think something is not right here. In case you do not know, the "application/octet-stream" [should be used for **binary** data](https://www.iana.org/assignments/media-types/application/octet-stream), not for JSON.

To reproduce, simply grab [your own script](https://amqp.opensuse.org/files/amqp_get_opensuse.py) and run it, changing its content like so:

```python
def callback(ch, method, properties, body):
    print(properties.content_type)
    print(" [x] %r:%r" % (method.routing_key, body))
```

At the end you will see that e.g. `opensuse.openqa.job.create` is `application/json` but e.g. `opensuse.obs.repo.build_started` is `application/octet-stream`, however both contain JSON body (as bytes).

@adrianschroeter @mlschroe why is that unfriendly attitude by closing issues by your colleagues? :disappointed: Can it be please a bit better handled here? Thanks in advance.

**Comment by hennevogel** [comment]
Ah so you compare *openqa* messages with OBS messages. Not OBS messages with OBS metrics. I understand now, you want us to add the "right" content type to our non-metric messages.

And not sure why you think closing issues is unfriendly. I can assure you it's not, it's just a button.

**Comment by isbm** [comment]
Actually from AMQP perspective there should be a way to detect what is the content of the whatever message. Say, if `user.login` is at all in a different format (as you already explained) that's totally fine (still would be nice to have them converted to a JSON for consistency). But if not, then it should not be `octet-stream` but something else. So then when you are listening to all messages, you know that you are picking up only the right ones and those others can be just skipped. But right now it is not possible. Hence I am asking to improve it here.

**Comment by hennevogel** [comment]
AFAIK AMQP doesn't really care about content type of messages, they leave this up to publisher/consumers. 

We also are *not* coordinating with other publishers on rabbit.opensuse.org about anything. So consuming *all* messages, in a consistent way, across all routing keys is not something anyone really cares about. Feel free to jump in, you seem to have a use case right?

What we can do here is to add a content type to the messages we publish.

**Comment by hennevogel** [comment]
So to summarize:

- Event's we send in `RabbitmqBus.send_to_bus` should include the `content_type: application/json` header.
- Metrics we send should include a `content_type: whatever/represents-influxdb-line-protocol-best` header

**Comment by isbm** [comment]
I would rephrase to:

- JSON messages should have content-type `application/json`
- Metrics from the influxdb should have `whatever/represents-influxdb-line-protocol-best` content type
